### PR TITLE
prevent index name collisions

### PR DESCRIPTION
**Pull request by Cuchulain**
Fix for bug on app enabling

### CODE REVIEW

**Comment by Cuchulain** [comment]
Bug in Scrutinizer test definition (SSL certificate problem: certificate has expired)

**Comment by Cuchulain** [comment]
With `curl -k` test passed, but not for nextcloud **master** branch.

**Comment by PhrozenByte** [comment]
Thank you for your contribution :+1:

> Fix for bug on app enabling

What bug? :thinking: Unless they didn't change something I missed, indices don't have to be unique across tables.

**Comment by Cuchulain** [comment]
> What bug? 🤔 Unless they didn't change something I missed, indices don't have to be unique across tables.

In last Nextcloud version pico_cms app can't be instaled, because index **user_id** already exists in NC core definitions.

**Comment by Cuchulain** [comment]
> ...indices don't have to be unique across tables.

I am using PostgreSQL, where index names is unique ;-)

**Review by PhrozenByte — CHANGES_REQUESTED** [review]
I see, thank you for the clarification :+1: This is indeed a bug then.

It's reproducible with SQLite, too. See https://scrutinizer-ci.com/g/nextcloud/cms_pico/inspections/3b9be198-9690-4fee-97bc-5e4d36e58087

Usually we don't edit existing migrations, because migrations run only once. We rather create a new migration addressing the issue. However, the `Version010000` migration will fail as soon as it is executed (i.e. later migrations won't come into play), so we must indeed edit the `Version010000` migration.

However, we still need an additional migration to also migrate existing installs. Otherwise old installs will use the `user_id` index, whereas new installs use `cms_pico_websites_user_id`. Please add another migration to change the index for existing installs, too.

File: .scrutinizer.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ build:
             - xdebug-2.9.8
       dependencies:
         override:
-          - curl -L -o ~/nextcloud.tar.bz2 https://download.nextcloud.com/server/$NEXTCLOUD.tar.bz2
+          - curl -k -L -o ~/nextcloud.tar.bz2 https://download.nextcloud.com/server/$NEXTCLOUD.tar.bz2
```
[the remark this answers is not in the document]
Comment (PhrozenByte) [reply .scrutinizer.yml:19]: I've just moved most Scrutinizer CI checks to GitHub Actions. Scrutinizer CI is now fully optional and used for analysis only. As soon as other issues occur due to their ancient software we'll remove Scrutinizer CI altogether...

**Comment by PhrozenByte** [comment]
Any updates @Cuchulain?

**Comment by PhrozenByte** [comment]
Fixed in dbe2b4a

Please try cms_pico's latest version v1.0.17.

**Pull request closed without merging**